tests/test_defaults.py

```diff
@@ -0,0 +1,85 @@
+"""
+Tests for Ralph's configuration loading
+"""
+
+from os import environ
+
+import pytest
+from click.testing import CliRunner
+
+import ralph.defaults
+from ralph.cli import cli
+from ralph.defaults import APP_DIR, config, load_config
+from ralph.exceptions import ConfigurationException
+
+
+# pylint: disable=invalid-name, unused-argument
+def test_defaults_config_default(fs, monkeypatch):
+    """Test that when no value is given, the default parameter is chosen"""
+
+    assert environ.get("RALPH_HISTORY_FILE", None) is None
+    assert config("RALPH_HISTORY_FILE", "history_default") == "history_default"
+
+
+# pylint: disable=invalid-name, unused-argument
+def test_defaults_config_file(fs, monkeypatch):
+    """Test that when the environment and command-line arguments don't specify a
+    value, the value from the config file is taken instead"""
+
+    config_mock = {"RALPH_HISTORY_FILE": "history_config_file"}
+
+    monkeypatch.setattr(ralph.defaults, "CONFIG", config_mock)
+
+    assert environ.get("RALPH_HISTORY_FILE", None) is None
+    assert config("RALPH_HISTORY_FILE", "history_default") == "history_config_file"
+
+
+# pylint: disable=invalid-name, unused-argument
+def test_defaults_config_env(fs, monkeypatch):
+    """Test that when the environment specifies a value, it is chosen instead of
+    the configuration file's value"""
+
+    config_mock = {"RALPH_HISTORY_FILE": "history_config_file"}
+
+    monkeypatch.setattr(ralph.defaults, "CONFIG", config_mock)
+    monkeypatch.setenv("RALPH_HISTORY_FILE", "history_env")
+
+    assert config("RALPH_HISTORY_FILE", "history_default") == "history_env"
+
+
+# pylint: disable=invalid-name, unused-argument
+def test_defaults_load_config_no_config(fs, monkeypatch):
+    """Test that loading when no configuration file is available results in
+    load_config returning None"""
+
+    runner = CliRunner()
+
+    fs.create_dir("/dev")
```

Comment:
I still don't get this.

Reply:
Neither do I unfortunately
Should we leave it like that ?